proposals/lip-0051.md

```diff
@@ -2447,7 +2448,9 @@ Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets fo
   * The `supplySubstore` array must be in ascending order of `tokenID`.
   * Across all elements of the `escrowSubstore` array, the `tokenID` must be unique (it appears at most once).
   * The `escrowSubstore` array must be in ascending order of `tokenID`.
-  * The `supportedTokensSubstore` array must be in lexicographical order of `chainID`.  For each entry of this array, the `supportedTokenIDs` array should be in lexicographic order.
+  * The `supportedTokensSubstore` array must adhere to one of the following two mutually exclusive conditions:
+    * Have a unique entry with `chainID` of length `0` and no other entries. This indicates that all tokens are supported, i.e., `chainID == ALL_SUPPORTED_TOKENS_KEY`. In this case, the `supportedTokenIds` should be an empty array.
+    * Have all entries with `chainID` property of length `CHAIN_ID_LENGTH`. The `supportedTokenIDs` array must be in lexicographic order, and the `supportedTokensSubstore` array must be in lexicographic order of `chainID`.
```

Comment:
In similar cases in other LIPs we usually require that all entries are unique, you could also add it here for consistency.